crates/transaction-pool/src/pool/txpool.rs

```diff
@@ -434,6 +434,11 @@ impl<T: TransactionOrdering> TxPool<T> {
         self.pending_pool.all()
     }
 
+    /// Returns the number of transactions from the pending sub-pool
+    pub(crate) fn pending_transactions_count(&self) -> usize {
+        self.pending_pool.all().count()
```

Comment:
I've optimized this a bit more so that this returns just .len() because now this does a bunch of Arc clones and this is equivalent to .len()